core/state_transition.go

```diff
@@ -465,7 +465,7 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 					"rewardsCount", len(discountConfig.Rewards))
 
 				// Check if the discount config is active and has a timestamp
-				if discountConfig.IsActive {
+				if discountConfig.IsActive && discountConfig.DiscountRate != nil && discountConfig.DiscountRate.Sign() > 0 {
```

Comment:
The `LFMDiscountConfigProvider.isValidConfig` does this checks already. Let's also add the checks in here, as it's were the discount applies and it is good to have.